tests/unit/charms/tls_certificates_interface/v2/test_tls_certificates_v2.py

```diff
@@ -414,3 +450,15 @@ def test_given_cert_and_private_key_when_generate_pfx_package_then_pfx_file_is_g
         initial_certificate=admin_operator_pem,
         initial_private_key=admin_operator_key_pem,
     )
+
+
+def test_given_matching_cert_for_csr_when_csr_matches_certificate_then_it_returns_true():
+    csr = EXAMPLE_CSR
+    certificate = EXAMPLE_CERT
+    assert csr_matches_certificate(csr, certificate) is True
+
+
+def test_given_non_matching_cert_for_csr_when_csr_matches_certificate_then_it_returns_false():
+    csr = EXAMPLE_CSR
+    certificate = "some random cert"
+    assert csr_matches_certificate(csr, certificate) is False
```

Comment:
We have 3 conditions that can make this method return False and here we're just validating 1 of them

Reply:
Added.